File: plataforma-web/web/backend/src/models/favoritos.js
const Sequelize = require('sequelize');
module.exports = function(sequelize, DataTypes) {
  return sequelize.define('favoritos', {
    id_favorito: {
      autoIncrement: true,
      autoIncrementIdentity: true,
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true
    },
    id_formando: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'utilizador',
        key: 'id_utilizador'
      }
    },
    id_curso: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'cursos',
        key: 'id_curso'
      }
    }
  }, {
    sequelize,
    tableName: 'favoritos',
    schema: 'public',
    timestamps: false,
    indexes: [
      {
        name: "pk_favoritos",
        unique: true,
        fields: [
          { name: "id_favorito" },
        ]
      },
    ]
  });
};

File: plataforma-mobile/mobile/lib/data/database/favoritos.dart
import 'package:sqflite/sqflite.dart';

Future<void> createFavoritosTable(Database db) async {
  await db.execute('''
    CREATE TABLE IF NOT EXISTS FAVORITOS (
      ID_FAVORITO INTEGER PRIMARY KEY AUTOINCREMENT,
      ID_FORMANDO INTEGER NOT NULL,
      ID_CURSO INTEGER NOT NULL,
      FOREIGN KEY (ID_FORMANDO) REFERENCES UTILIZADOR(ID_UTILIZADOR),
      FOREIGN KEY (ID_CURSO) REFERENCES CURSOS(ID_CURSO)
    );
  ''');
}